### ISSUE TITLE
[BUG] - AnalysisCriterion are being calculated on bar closing price and not on order price/amount

### ISSUE THREAD

**Issue by GxtPerfectoM** [ta4j/ta4j#648]
In some AnalysisCriterions the calculate method is using the bar-series bars closing price (By order index) for doing the calculations instead of using the order price/amount it-self.

For example: NumberOfBreakEvenTradesCriterion.
The current code:
Num exitPrice = series.getBar(trade.getExit().getIndex()).getClosePrice();
Num entryPrice = series.getBar(trade.getEntry().getIndex()).getClosePrice();
Num profit = exitPrice.minus(entryPrice).multipliedBy(trade.getExit().getAmount());
return profit.isZero();

Instead it should be:
Num exitPrice = trade.getExit().getPricePerAsset();
Num entryPrice = trade.getEntry().getPricePerAsset();
Num profit = exitPrice.minus(entryPrice).multipliedBy(trade.getExit().getAmount());
return profit.isZero();

This behavior can be seen in more AnalysisCriterions implementation.

**Comment by team172011** [comment]
@GxtPerfectoM this is a known issue already addressed in #23

**Comment by team172011** [comment]
already resolved by using the Position#getProfit function

```java
    private boolean isBreakEvenTrade(Position position) {
        if (position.isClosed()) {
            return position.getProfit().isZero();
        }
        return false;
    }
```
